### ISSUE TITLE
TS-993004: Unable to import symbol NgIf

### ISSUE THREAD

**Issue by ky-one** [angular/angular#59247]
### Which @angular/* package(s) are the source of the bug?

common

### Is this a regression?

Yes

### Description

After upgrading to latest version I get this error:

```
X [ERROR] TS-993004: Unable to import symbol NgIf.
  The symbol is not exported from .../node_modules/@angular/common/index.d.ts (module '@angular/common'). [plugin angular-compiler]

    src/app/components/user/profile/profile.component.ngtypecheck.ts:0:0:
      0 │
        ╵ ^

  The symbol is declared here.

    node_modules/@angular/common/index.d.ts:2291:0:
      2291 │ export declare class NgIf<T = unknown> {
           ╵ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The component look like this:

``` html
<m-table *ngIf="user" very basic notCelled>
    <tr>
        <th class="collapsing">Benutzer</th>
        <td>{{ user.name }} ({{ user.displayName }})</td>
    </tr>
    <tr>
        <th>Passwort</th>
        <td>
            <m-button primary (click)="showChangePassword()">ändern</m-button>
        </td>
    </tr>
    <tr>
        <th>E-Mail</th>
        <td>
            {{ user.email }}
            <m-button disabled>ändern</m-button>
        </td>
    </tr>
    <tr>
        <th>Rechte</th>
        <td>
            <div *ngFor="let role of user.roles">{{ role }}</div>
        </td>
    </tr>
</m-table>

<m-loader *ngIf="!user" />

```

``` ts
import { NgForOf, NgIf } from '@angular/common';
import { Component, OnInit } from '@angular/core';
import { AuthenticationService, UserModel } from '@ky/authentication';
import { ButtonComponent, LoaderComponent, TableComponent } from '@mantic-ui/angular';

@Component({
    selector: 'app-profile',
    standalone: true,
    imports: [NgIf, NgForOf, TableComponent, ButtonComponent, LoaderComponent],
    templateUrl: './profile.component.html',
    styleUrls: ['./profile.component.scss']
})
export class ProfileComponent implements OnInit {
    protected user?: UserModel;
    protected isChangePasswordVisible = false;

    public constructor(
        private readonly authenticationService: AuthenticationService
    ) { }

    public ngOnInit(): void {
        this.authenticationService.getUser().subscribe(user => this.user = user);
    }

    protected showChangePassword(): void {
        this.isChangePasswordVisible = true;
    }

    protected hideChangePassword(): void {
        this.isChangePasswordVisible = false;
    }
}

```

The code is unchanged. I only upgraded from angular 19.0.0 to latest version

### Please provide the exception or error you saw

```true
X [ERROR] TS-993004: Unable to import symbol NgIf.
  The symbol is not exported from .../node_modules/@angular/common/index.d.ts (module '@angular/common'). [plugin angular-compiler]

    src/app/components/user/profile/profile.component.ngtypecheck.ts:0:0:
      0 │
        ╵ ^

  The symbol is declared here.

    node_modules/@angular/common/index.d.ts:2291:0:
      2291 │ export declare class NgIf<T = unknown> {
           ╵ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 19.0.6
Node: 20.14.0
Package Manager: npm 10.9.0
OS: win32 x64

Angular: 19.0.5
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
... service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1900.6
@angular-devkit/build-angular   19.0.6
@angular-devkit/core            19.0.6
@angular-devkit/schematics      19.0.6
@angular/cli                    19.0.6
@schematics/angular             19.0.6
rxjs                            7.8.1
typescript                      5.6.3
zone.js                         0.15.0
```

**Comment by JeanMeche** [comment]
Can you delete your `.angular` cache and retry ?

**Comment by ky-one** [comment]
@JeanMeche sadly, this does not change anything

**Comment by JeanMeche** [comment]
Are you able to reproduce it on a stackblitz/ demo github repo  ?

**Comment by ky-one** [comment]
Ok, I solved it.
The error message was totally misleading. It looks like the error was not in ProfileComponent, it was in a component used in ProfileComponent. 
The components starting with m- are from an external library. This library is normal installed in node_modules, but to debug an error I mapped the library via tsconfig.app.json compileOptions paths to the source directory. After removing the paths, everything works now.
